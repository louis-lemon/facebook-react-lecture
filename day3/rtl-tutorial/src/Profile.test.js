import React from 'react';
import { render } from 'react-testing-library';
import Profile from './Profile';

describe('<Profile />', () => {
  it('matches snapshot', () => {
    const utils = render(<Profile username="louis" name="tak" />);
    expect(utils.container).toMatchSnapshot();
  });

  it('shows the props correctly', () => {
    const utils = render(<Profile username="louis" name="tak" />);
    utils.getByText('louis'); // louis 라는 텍스트를 가진 엘리먼트가 있는지 확인
    utils.getByText('(tak)'); // (tak) 이라는 텍스트를 가진 엘리먼트가 있는지 확인
    utils.getByText(/louis/); // 정규식 /louis/ 을 통과하는 엘리먼트가 있는지 확인
  });


});
