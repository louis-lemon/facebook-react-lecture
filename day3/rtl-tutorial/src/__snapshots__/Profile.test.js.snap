// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<div>
  <div>
    <b>
      louis
    </b>
    Â 
    <span>
      (
      tak
      )
    </span>
  </div>
</div>
`;
